src/microbe_stage/editor/action_data/ColourActionData.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Godot;
+
+[JSONAlwaysDynamicType]
+public class ColourActionData : EditorCombinableActionData
+{
+    public Color NewColour;
+    public Color PreviousColour;
+
+    public ColourActionData(Color newColour, Color previousColour)
+    {
+        NewColour = newColour;
+        PreviousColour = previousColour;
+    }
+
+    public override bool WantsMergeWith(CombinableActionData other)
+    {
+        return other is ColourActionData;
+    }
+
+    protected override int CalculateCostInternal()
+    {
+        // Changing membrane color has no cost
+        return 0;
+    }
+
+    protected override CombinableActionData CombineGuaranteed(CombinableActionData other)
+    {
+        var colourChangeActionData = (ColourActionData)other;
+
+        if (PreviousColour.IsEqualApprox(colourChangeActionData.NewColour))
+            return new ColourActionData(NewColour, colourChangeActionData.PreviousColour);
+
+        return new ColourActionData(colourChangeActionData.NewColour, PreviousColour);
+    }
+
+    protected override void MergeGuaranteed(CombinableActionData other)
+    {
+        var colourChangeActionData = (ColourActionData)other;
+
+        if (PreviousColour.IsEqualApprox(colourChangeActionData.NewColour))
+        {
+            // Handle cancels out
+            if (NewColour.IsEqualApprox(colourChangeActionData.PreviousColour))
+            {
+                NewColour = colourChangeActionData.NewColour;
+                return;
+            }
+
+            PreviousColour = colourChangeActionData.PreviousColour;
+            return;
+        }
+
+        NewColour = colourChangeActionData.NewColour;
+    }
+
+    protected override ActionInterferenceMode GetInterferenceModeWithGuaranteed(CombinableActionData other)
```

Comment:
These methods look like they are in different order than other actions, please reorder them to be in a consistent order.

Reply:
Right, was put this way by my IDE.

Reply:
Fixed.